stdlib/REPL/test/replcompletions.jl

```diff
@@ -1896,3 +1896,17 @@ let s = "Issue49892(fal"
         @test n in c
     end
 end
+
+@testset "public but non-exported symbols only complete qualified (#51331)" begin
+    c, r, res = test_complete("ispub")
+    @test res
```

Comment:
What is `res` actually testing?

Reply:
I think it's something about the type of completion, perhaps weather or not it is ever sensible to display the completion inline. For example, IICU, we never display argument hints inline.
```julia
julia> completions("summ", 4)
(REPL.REPLCompletions.Completion[REPL.REPLCompletions.ModuleCompletion(Base, "summary"), REPL.REPLCompletions.ModuleCompletion(Base, "summarysize")], 1:4, true)

julia> completions("summary(", 8)
(REPL.REPLCompletions.Completion[REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), Base.JuliaSyntax.SyntaxHead}, summary(head::Base.JuliaSyntax.SyntaxHead) @ Base.JuliaSyntax ~/.julia/dev/julia/base/JuliaSyntax/src/parse_stream.jl:98), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), Base.JuliaSyntax.GreenNode}, summary(node::Base.JuliaSyntax.GreenNode) @ Base.JuliaSyntax ~/.julia/dev/julia/base/JuliaSyntax/src/green_tree.jl:39), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), IO, Function}, summary(io::IO, f::Function) @ Base show.jl:3111), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), IO, AbstractDict}, summary(io::IO, t::AbstractDict) @ Base abstractdict.jl:33), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), IO, T} where T<:Union{Base.KeySet, Base.ValueIterator}, summary(io::IO, iter::T) where T<:Union{KeySet, ValueIterator} @ Base abstractdict.jl:51), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), IO, AbstractString}, summary(io::IO, s::AbstractString) @ Base strings/basic.jl:234), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), IO, AbstractSet}, summary(io::IO, t::AbstractSet) @ Base show.jl:214), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), IO, AbstractArray}, summary(io::IO, a::AbstractArray) @ Base show.jl:3099), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), IO, Any}, summary(io::IO, x) @ Base show.jl:3081), REPL.REPLCompletions.MethodCompletion(Tuple{typeof(summary), Any}, summary(x) @ Base show.jl:3082)], 1:7, false)

julia> summary(
summary(head::Base.JuliaSyntax.SyntaxHead) @ Base.JuliaSyntax ~/.julia/dev/julia/base/JuliaSyntax/src/parse_stream.jl:98
summary(node::Base.JuliaSyntax.GreenNode) @ Base.JuliaSyntax ~/.julia/dev/julia/base/JuliaSyntax/src/green_tree.jl:39
summary(io::IO, f::Function) @ Base show.jl:3111
summary(io::IO, t::AbstractDict) @ Base abstractdict.jl:33
summary(io::IO, iter::T) where T<:Union{KeySet, ValueIterator} @ Base abstractdict.jl:51
summary(io::IO, s::AbstractString) @ Base strings/basic.jl:234
summary(io::IO, t::AbstractSet) @ Base show.jl:214
summary(io::IO, a::AbstractArray) @ Base show.jl:3099
summary(io::IO, x) @ Base show.jl:3081
summary(x) @ Base show.jl:3082
julia> summary(
```